rust/src/core/map_session.rs

```diff
@@ -262,9 +261,11 @@ impl MapSession {
         self.redraw().await.inspect_err(|err| error!("{err}"));
     }
 
-    pub async fn terminate(self: Arc<Self>) {
+    pub async fn terminate(self: Arc<Self>) -> Option<FlutterCtx> {
         self.is_alive.store(false, Ordering::SeqCst);
 
+        let flctx = self.flutter_ctx.write().take();
```

Comment:
I'd move this out of the impl and call drop on the platform thread here instead of in the api